docs/diagrams/ContactListSelectionChangeDiagram.puml

```diff
@@ -0,0 +1,52 @@
+@startuml
+!include style.puml
+
+box UI UI_COLOR_T1
+participant ":PersonListView" as PersonListView UI_COLOR
+participant ":PersonListPanel" as PersonListPanel UI_COLOR
+participant ":ContactDetailPanel" as ContactDetailPanel UI_COLOR
+participant ":ResultDisplay" as ResultDisplay UI_COLOR
+end box
+
+box Logic LOGIC_COLOR_T1
+participant ":Logic" as Logic LOGIC_COLOR
+end box
+
+[-> PersonListView : selectItem()
+activate PersonListView
+
+PersonListView -> PersonListPanel : handleSelectionChange()
+activate PersonListPanel
+
+PersonListPanel -> Logic : setCurrentlySelectedPerson()
+activate Logic
+
+Logic -> ContactDetailPanel : handleSelectionChange(newPerson)
+deactivate Logic
+activate ContactDetailPanel
+
+ContactDetailPanel -> ContactDetailPanel : setPanel(new contactDetailCard(newPerson))
+activate ContactDetailPanel
+
+ContactDetailPanel -> ResultDisplay : setFeedbackToUser("Details of [name] shown!")
```

Comment:
![image](https://user-images.githubusercontent.com/72195240/162573940-f163c4a2-0824-427b-a060-4750b78a7f63.png)

change this section to this?
```
PersonListPanel -> Logic : setCurrentlySelectedPerson(person)
activate Logic

Logic -> : set observable
deactivate Logic 

ContactDetailPanel <- : handleSelectionChange(person)
deactivate Logic
activate ContactDetailPanel

ContactDetailPanel -> ContactDetailPanel : setPanel(new contactDetailCard(person))
```

Reply:
basically `Logic` should forward the setter to outside its class, and `handleSelectionChange` is called by an `Observable` outside of `Logic`